core/src/main/java/com/yahoo/ycsb/workloads/CoreWorkload.java

```diff
@@ -525,244 +559,238 @@ private String buildDeterministicValue(String key, String fieldkey) {
     return sb.toString();
   }
 
-	/**
-	 * Do one insert operation. Because it will be called concurrently from multiple client threads, this 
-	 * function must be thread safe. However, avoid synchronized, or the threads will block waiting for each 
-	 * other, and it will be difficult to reach the target throughput. Ideally, this function would have no side
-	 * effects other than DB operations.
-	 */
-	public boolean doInsert(DB db, Object threadstate)
-	{
-		int keynum=keysequence.nextInt();
-		String dbkey = buildKeyName(keynum);
-		HashMap<String, ByteIterator> values = buildValues(dbkey);
-		if (db.insert(table,dbkey,values).equals(Status.OK))
-			return true;
-		else
-			return false;
-	}
-
-	/**
-	 * Do one transaction operation. Because it will be called concurrently from multiple client threads, this 
-	 * function must be thread safe. However, avoid synchronized, or the threads will block waiting for each 
-	 * other, and it will be difficult to reach the target throughput. Ideally, this function would have no side
-	 * effects other than DB operations.
-	 */
-	public boolean doTransaction(DB db, Object threadstate)
-	{
-		String op=operationchooser.nextString();
-
-		if (op.compareTo("READ")==0)
-		{
-			doTransactionRead(db);
-		}
-		else if (op.compareTo("UPDATE")==0)
-		{
-			doTransactionUpdate(db);
-		}
-		else if (op.compareTo("INSERT")==0)
-		{
-			doTransactionInsert(db);
-		}
-		else if (op.compareTo("SCAN")==0)
-		{
-			doTransactionScan(db);
-		}
-		else
-		{
-			doTransactionReadModifyWrite(db);
-		}
-		
-		return true;
-	}
+  /**
+   * Do one insert operation. Because it will be called concurrently from multiple client threads,
+   * this function must be thread safe. However, avoid synchronized, or the threads will block waiting
+   * for each other, and it will be difficult to reach the target throughput. Ideally, this function would
+   * have no side effects other than DB operations.
+   */
+  public boolean doInsert(DB db, Object threadstate) {
+    int keynum = keysequence.nextInt();
+    String dbkey = buildKeyName(keynum);
+    HashMap<String, ByteIterator> values = buildValues(dbkey);
+
+    Status status;
```

Comment:
After thinking about this for a bit, I'd prefer to keep the retry in the CoreWorkload class because the need for retry is largely driven by the design of the workload, e.g., a future workload may be able to ignore failed insertions during the load phase (and subsequently avoid touching those records that were not inserted during its run phase). It'd be quite hard to do such for the core workload in its current form, because doing so would require state tracking across runs in the random key number generator. A simpler and more robust way for the core workload is to just retry those failed insertions.

I like the idea of prefixing the retry limit property with "core_workload" so it differentiates from others. I just pushed a change for that. Thanks for the suggestion!